src/utils/array/index.ts

```diff
@@ -18,6 +18,18 @@ export const toggleArray = <T>(value: T, array: T[]) => {
   return tempArray;
 };
 
+/**
+ * deduplicatedAppend takes in an array of values regardless of type and a new value and safely inserts the value if it doesn't exist in the array.
+ * if it does exist it will do nothing
+ * @param value The value to insert into the array.
+ * @param array The array to insert the value into.
+ */
+export const deduplicatedAppend = <T>(value: T, array: T[]) => {
+  let tempArray = new Set([...array]);
```

Comment:
`[...array]` = `array`
```suggestion
  let tempArray = new Set(array);
```